### ISSUE TITLE
Win7 builds failing with ""GetTargetMachineInfo" task failed unexpectedly"

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#7294]
e.g. http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_win7_debug/141/console
```
d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018: The "GetTargetMachineInfo" task failed unexpectedly. [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018: System.DllNotFoundException: Unable to load DLL 'api-ms-win-core-sysinfo-l1-2-0.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E) [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018:    at Interop.mincore.GetNativeSystemInfo(SYSTEM_INFO& lpSystemInfo) [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018:    at System.Runtime.InteropServices.RuntimeInformation.get_OSArchitecture() [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.GetTargetMachineInfo.Execute() [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
18:12:03 d:\j\workspace\outerloop_win---9c9e7d59\Tools\publishtest.targets(77,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [d:\j\workspace\outerloop_win---9c9e7d59\src\Common\tests\Common.Tests.csproj]
```

**Comment by Priya91** [comment]
This version of buildtools is not there in rc2 branch, pushing this bug to rtm. On win7 we are using the net45 lib, and the win api sets are not being copied over to the Tools\net45. Explicitly specifying the package ```Microsoft.NETCore.Windows.ApiSets``` is also not fixing this. Investigating.

**Comment by Priya91** [comment]
Closing buildtools PR, closed this issue, as it was tagged there. Reopening again.

**Comment by Priya91** [comment]
fixed.

**Comment by stephentoub** [comment]
This is happening again, e.g.
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/outerloop_win7_release/9/consoleText

**Comment by Priya91** [comment]
This seems to be happening from the cli update.. The dotnet.exe publish is not longer copying all deps locally for net45. It seems like now, for dnxcore50, it is creating a .deps.json file which lists the dependencies for dnxcore50, but there is no .deps.json for net45. Filed an issue on dotnetcli repo.

**Comment by Priya91** [comment]
This is fixed in #8252
